### PR TITLE
Run single host multi GPU tests in the CI.

### PR DESCRIPTION
**Pull request by vanbasten23**
This pr is intended to run the single host multi GPU tests in the CI.

### CODE REVIEW

**Review by jonb377 — APPROVED** [review]
LGTM

**Review by will-cromar — APPROVED** [review]
Thanks!

File: test/pjrt/test_runtime_gpu.py [hunk 1]
```diff
@@ -166,6 +177,8 @@ def _reduce_scatter(pin_layout):
 
     return out.cpu().numpy()
 
+  # 2023-08-02 04:16:36.520884: F external/xla/xla/service/layout_assignment.cc:157] Check failed: ShapeUtil::Compatible(shape_layout.shape(), instruction->operand(operand_no)->shape()) f32[1]{0} is not compatible with f32[2]{0} (for operand 0 of instruction %reduce-scatter.10 = f32[1]{0} reduce-scatter(f32[2]{0} %add.5), replica_groups={}, constrain_layout=true, dimensions={0}, to_apply=%AddComputation.6)
```
[the remark this answers is not in the document]
Comment (vanbasten23) [reply test/pjrt/test_runtime_gpu.py:180]: Right. Running it on one GPU works:
```
root@4a7f03f198d2:/ansible# GPU_NUM_DEVICES=1 PJRT_DEVICE=GPU python pytorch/xla/test/pjrt/test_runtime_gpu.py TestExperimentalPjrtGpu.test_reduce_scatter_pinned
Running tests under Python 3.8.17: /usr/local/bin/python
[ RUN      ] TestExperimentalPjrtGpu.test_reduce_scatter_pinned
[       OK ] TestExperimentalPjrtGpu.test_reduce_scatter_pinned
----------------------------------------------------------------------
Ran 1 test in 9.790s

OK
```

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**